General UI changes

BMartinos opened the pull request:
This is just some general style tweaks to change the look and feel of the console.

BMartinos commented:
Hey @devcritter @rcrichton @debonair if you guys have some time then take a look at the few style additions ive been working on. I think this gives the console a better look and feel and moves away from the default boostrap Admin theme. Comments/suggestions welcome :)

devcritter commented:
Thanks @BMartinos I like the changes! It uses more of the screen space as well.

I just thought though that the header and footer dropshadows didn't look to great with these edits, maybe we should try minimize them? Or maybe even experiment with getting rid of the shadows completely?
![screen shot 2015-04-09 at 11 29 38](https://cloud.githubusercontent.com/assets/1872071/7064155/a8ee61ea-deac-11e4-9440-338802232765.png)

rcrichton commented:
@BMartinos I like the changes to. I agree with Hannes, perhaps we should play with the dropshadows a bit and see how it looks. Making them very subtle may looks cool.

rcrichton commented:
Also, for some inspiration it may be good to look at google's material design language http://www.google.com/design/spec/material-design/introduction.html#

It seems you are kind of going in this direction already so you may have already seen this.

BMartinos commented:
Cool ill have a look at those box-shadows so that they are more subtitle. Thansk for the link @rcrichton, that is really cool. Haven't seen that before but def in the direction i was thinking of and will be sourcing a few of those styles ideas.

BMartinos commented:
Hey guys. Ive made some small changes regarding the box-shadows. I think this does look much better with a softer shadow at the header and footer. Let me know if you feel anything else needs to change.

devcritter commented:
Thanks, looks good to me. Maybe we can just try one last thing before merging and see what it looks like without any drop shadows?

BMartinos commented:
Cool ive made those changes to remove the box-shadows. Have a look and let me know which you think is better.

rcrichton commented:
Hmm. I like the softer shadow. However, the pure flat looks like it could work with a few more edits. I feel like there is something missing from both though, but I'm not sure what else could be done. Perhaps we should have a meeting sometime this week while we are all together and brain storm ideas about would would be best. I'm not very good at UI design but perhaps if we put our heads together we can come up with something even better.

devcritter commented:
I actually quite like it without the drop shadow. But that's a great idea, let's meet around this. Maybe even later today?

BMartinos commented:
Yeah that sounds like a good idea to me. As it is now im more in favour of the softer box-shadows but do agree that something does feel a bit missing from both. Having a discussion will help a lot to get in the right direction and what works better.

BMartinos commented:
After our meeting earlier we have a few changes proposed to the UI. These will be added to a new issue shortly. Should these UI changes fall under the the new changes branch or can this be merged in in the meantime?

rcrichton commented:
@BMartinos maybe its better to create new branches for the new UI refactoring from our discussions. (or reuse this branch) Then we can merge once we are happy with the look?

BMartinos commented:
Cool sounds good to me. Im currently using this existing branch for the general UI changes. It might be easier to do the transactions changes in this branch as well

BMartinos commented:
This PR relates to issue: https://github.com/jembi/openhim-console/issues/294

BMartinos commented:
Hey @rcrichton @devcritter @debonair i have made all the general UI changes. When you guys have a chance take and look. Thanks

rcrichton commented:
Hey @BMartinos just having a look at this visually. A few thoughts:

* Maybe use a dark icon for the tooltips: I like 'icon-question-sign' from http://marcoceppi.github.io/bootstrap-glyphicons/ the colour sticks out too much
* I think we can get rid of the toggle UI setting button on the header, I think it makes sense to always have them there.
* I don't like the grey background on the selected item on the sidebar, it blends into the background, maybe we can use coloured text here instead (the green?)
* It would be great if you could add a shadow on the bottom of the header but only when the user has scrolled down a bit (this indicates that some stuff is under the header)
* I see the first route that you set doesn't default to the primary, could you add this?
* One other thing is the user menu and the menu that you get on mobile have a rounded look that conflicts with the rest of the changes

Overall, its looks great. Good work so far.

BMartinos commented:
Cool thanks for the feedback @rcrichton will start looking at those changes. Regarding the UI settings button, I added that in because the tooltip icons might be a bit over-whelming on certain pages ( audits with advanced filters open ). This was just as a quick way to toggle it on and off. I have added this as a general user setting so they can manage it there.

BMartinos commented:
Hey @rcrichton i have made those changes you mentioned. Ive also started working on the transactions changes and unfortunately added them into this branch. Most of them are completed as well. Might just need a few small tweaks

rcrichton commented:
@BMartinos this is looking awesome! Let me know when you have done the transaction changes and I'll review it more than just visually.

BMartinos commented:
Hey @rcrichton i have completed all the requested UI changes. Do you mind review this for me again. Thanks

rcrichton commented:
Hey @BMartinos, I'm not too sure about the transaction page. I think the boxes [here](http://www.awesomescreenshot.com/image/135703/98cc450849833a35dccca5155840a4c4) should rather be one table instead of separate boxes. It looks a bit disjointed I think.

rcrichton commented:
I  think they should be a table with 2 columns similar to before

rcrichton commented on app/index.html:
```diff
@@ -39,25 +39,24 @@
-                          </li>
-                        </ul>
-                      </nav>
+                  <div class="navbar navbar-inverse" role=" banner" style="float: right;">
```
Perhaps we should remove all inline styles and do it all is css?

BMartinos replied:
Correct, I added inline style at first as its a bit easier to get the layouts right. Must still move most of the styles into CSS classes

rcrichton commented on app/scripts/controllers/tasks.js:
```diff
@@ -4,12 +4,20 @@
+
+    var tasksInterval = $interval(function() {
+      $scope.refreshTasksList();
+    }, 5000);
```
This seems to make the task list 'flash' every 5s. If we could make this happen in the background and then just load in the new data it would be ideal. Else, maybe we remove this until we can figure that out.

BMartinos replied:
Will look into this. I might have to remove the loading icon then and make the status bars static and not animate

rcrichton commented on app/styles/main.css:
```diff
@@ -256,10 +258,10 @@ footer .copy{
    border-radius:10px;
    padding:10px;
-   border-left:1px solid #eee;
+   /*border-left:1px solid #eee;
```
Remove this commented out css

rcrichton commented on app/styles/main.css:
```diff
@@ -364,9 +366,9 @@ footer .copy{
-  background: #fff;
-  color: #555;
-   border-bottom:0px;
+  /*background: #eff0f3;*/
```
remove

rcrichton commented on app/views/profile.html:
```diff
@@ -46,7 +46,10 @@
               <div class="form-group" ng-class="{ 'has-error' : ngError.msisdn }">
-                <label tooltip-html-unsafe="Accepted format: <br />27123456789 <br />( 5 - 15 digits )">MSISDN</label>
+                <label>
+                  MSISDN
```
Maybe label this 'Phone number (MSISDN format)'

rcrichton commented on app/views/profile.html:
```diff
@@ -81,13 +84,15 @@
-                  <label tooltip="Receive Daily Channel Reports">
+                  <label>
                     <input type="checkbox" ng-model="user.dailyReport" /> Receive Daily Channel Reports?
+                    <i ng-show="uiSettings.showTooltips" tooltip-placement="top" tooltip="Receive Daily Channel Reports" class="glyphicon glyphicon-question-sign tooltip-wide"></i>
```
Maybe make the tooltip read: 'Receive daily channel reports that give a summary of the transactions run through channel that you can view.'

rcrichton commented on app/views/profile.html:
```diff
@@ -81,13 +84,15 @@
-                  <label tooltip="Receive Weekly Channel Reports">
+                  <label>
                     <input type="checkbox" ng-model="user.weeklyReport" /> Receive Weekly Channel Reports?
+                    <i ng-show="uiSettings.showTooltips" tooltip-placement="top" tooltip="Receive Weekly Channel Reports" class="glyphicon glyphicon-question-sign tooltip-wide"></i>
```
Same as above, but weekly

rcrichton commented on app/views/setPassword.html:
```diff
@@ -38,7 +38,10 @@
               <div class="form-group" ng-class="{ 'has-error' : ngError.msisdn }">
-                <label tooltip-html-unsafe="Accepted format: <br />27123456789 <br />( 5 - 15 digits )">MSISDN</label>
+                <label>
+                  MSISDN
```
Maybe label this 'Phone number (MSISDN format)'

rcrichton commented on app/views/transactionsAddReqResModal.html:
```diff
@@ -1,127 +1,180 @@
+<style>
```
Should we not put this is a css file?

BMartinos replied:
Ideally yes. Problem with putting this into the stylesheet is that it will apply for ALL modals. By doing it the way i did it with these modals it injects those styles and overrides the defaults to make the modals full height/width. This is only need on the transaction modals.

I can look at finding a better way for this

rcrichton replied:
Ok, it not too important for now, but ideally if we find a better way it would be better.

rcrichton commented on app/views/transactionsAddReqResModal.html:
```diff
@@ -1,127 +1,180 @@
+
 <div class="modal-header">
-  <button type="button" class="close" ng-click="cancel()" aria-hidden="true">&times;</button>
+  <button type="button" class="btn btn-default" ng-click="cancel()" style="float: right;">Close</button>
```
Move inline css to file?

rcrichton commented on app/views/transactionsBodyModal.html:
```diff
@@ -1,5 +1,19 @@
+<style>
```
Maybe move this to a css file instead

rcrichton commented on app/views/usersmodal.html:
```diff
@@ -36,7 +36,10 @@ <h4 class="modal-title" id="addUserLabel">
       <div class="form-group" ng-class="{ 'has-error' : ngError.msisdn }">
-        <label tooltip-html-unsafe="Accepted format: <br />27123456789 <br />( 5 - 15 digits )">MSISDN</label>
+        <label>
+          MSISDN
```
Maybe label this 'Phone number (MSISDN format)'

rcrichton commented on app/views/usersmodal.html:
```diff
@@ -69,13 +72,15 @@ <h4 class="modal-title" id="addUserLabel">
-          <label tooltip="Receive Daily Channel Reports">
+          <label>
             <input type="checkbox" ng-model="user.dailyReport" /> Receive Daily Channel Reports?
+            <i ng-show="uiSettings.showTooltips" tooltip-placement="top" tooltip="Receive Daily Channel Reports" class="glyphicon glyphicon-question-sign tooltip-wide"></i>
```
Same comment as above for profile

rcrichton commented on app/views/usersmodal.html:
```diff
@@ -69,13 +72,15 @@ <h4 class="modal-title" id="addUserLabel">
-          <label tooltip="Receive Weekly Channel Reports">
+          <label>
             <input type="checkbox" ng-model="user.weeklyReport" /> Receive Weekly Channel Reports?
+            <i ng-show="uiSettings.showTooltips" tooltip-placement="top" tooltip="Receive Weekly Channel Reports" class="glyphicon glyphicon-question-sign tooltip-wide"></i>
```
Same comment as above for profile

rcrichton commented:
Hey @BMartinos, looking great! Some comments above.

BMartinos commented:
Hey @rcrichton i have completed all of the comments mentioned except for removing all inline styles. I started with the but it will take a bit longer than i expected as there are a few places this needs to be corrected

rcrichton commented:
@BMartinos ok that's cool. I don't think we need to fix all that now. I'm having a look now.

rcrichton commented:
Hey @BMartinos all looks good, just one last thing: When I load more tasks with the button at the bottom, the reload set it back to 10.

BMartinos commented:
hmmm that is a bit of a problem. I didnt notice that as i had less than 10 reruns... Let me see if i can think of a way around this. Otherwise we will need to remove it for now

BMartinos commented:
Removed the Tasks auto refresh functionality. Opened a new issue for adding in the tasks auto refresh as this is a bit more complex as initially thought. @rcrichton do you mind just check all corrections for me again. Thanks

rcrichton commented:
Hey @BMartinos All looks good! :+1: Our console now looks pretty modern!

BMartinos commented:
WhooHoo Console got a bit of a make over!! :)
